src/Database/LSMTree/Normal.hs

```diff
@@ -425,12 +425,19 @@ open (Session sesh) snap = TableHandle <$> Internal.open sesh label snap
 -- memory and disk cost will be the same as if each table were entirely
 -- independent.
 --
--- NOTE: duplication create a new table handle, which should be closed when no
+-- NOTE: duplication creates a new table handle, which should be closed when no
 -- longer needed.
 --
+-- [Exception safety]: it is recommended to run this function with asynchronous
```

Comment:
This is true but it's probably a point we can make more generally, e.g. in the top level docs and then refer to it from the allocating actions. It's true of all resource-allocating actions that need a corresponding cleanup: new, open, duplicate. All of them should be handled carefully wrt async exceptions. If one uses `bracket` then this is handled automatically (since bracket masks exceptions during allocation and cleanup phases).

Reply:
Sounds reasonable, I'll change that